linkd-lib/src/cfg.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{
+    convert::TryFrom,
+    io,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4, ToSocketAddrs as _},
+    time::Duration,
+};
+
+use anyhow::Result;
+use thrussh_agent::client::ClientStream;
+use tokio::{
+    fs::File,
+    io::{stdin, AsyncReadExt as _},
+    time::{error::Elapsed, timeout},
+};
+use tracing::warn;
+
+use librad::{
+    crypto::{BoxedSigner, IntoSecretKeyError},
+    git::storage,
+    keystore::SecretKeyExt as _,
+    net,
+    net::{discovery, peer::Config as PeerConfig},
+    profile::{Profile, RadHome},
+    SecretKey,
+};
+use rad_clib::keys;
+
+use crate::args;
+
+mod seed;
+pub use seed::{Seed, Seeds};
+
+lazy_static::lazy_static! {
+    /// Localhost binding to any available port, i.e. `127.0.0.1:0`.
+    pub static ref LOCALHOST_ANY: SocketAddr =
+        SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 0));
```

Comment:
`([0,0,0,0], 0).into()`?

Reply:
That’s not a good default, either. I’d rather mandate that the bind address must be specified explicitly, and provide shortcuts such as “any”, “localhost”

Reply:
bd8ee1b